core/services/collections/Collection.php

```diff
@@ -35,20 +35,64 @@ class Collection extends SplObjectStorage implements CollectionInterface
      */
     protected $collection_interface;
 
+    /**
+     * a short dash separated string describing the contents of this collection
+     *
+     * @type string $collection_identifier
+     */
+    protected $collection_name;
```

Comment:
I realize that the comments say that this property is for describing the contents of this collection.  However, I think it'd also be useful to indicate in the phpdocs that the collection identifier is derived in prat from this property as well.